src/Utils/citesupport.ts

```diff
@@ -0,0 +1,453 @@
+      citation: citation,
+      preCitations: preCitations,
+      postCitations: postCitations,
+    });
+  }
+
+  getBibliography(): void {
+    if (!this.config.processorReady) return;
+    this.debug("getBibliography()");
+    this.config.processorReady = false;
+    this.worker.postMessage({
+      command: "getBibliography",
+    });
+  }
+
+  /**
+   *    Converts the array returned by the processor `rebuildProcessor()` method
+   *    to the form digested by our own `setCitations()` method.
+   *
+   *    rebuildData has this structure:
+   *    [<citation_id>, <note_number>, <citation_string>]
+   *
+   *    setCitations() wants this structure:
+   *    [<citation_index>, <citation_string>, <citation_id>]
+   *
+   *    @param {Object[]} rebuildData An array of values for insertion of citations into a document
+   *    @return {Object[]}
+   */
+  convertRebuildDataToCitationData(rebuildData: object[]): object[] {
+    if (!rebuildData) return null;
+    this.debug("convertRebuildDataToCitationData()");
+    var citationData = rebuildData.map((obj) => [0, obj[2], obj[0]]);
+    for (var i = 0; i < citationData.length; i++) {
+      citationData[i][0] = i;
+    }
+    return citationData;
+  }
+
+  /**
+   *   Function to be run immediately after document has been loaded, and
+   *   before any editing operations.
+   *   @return {void}
+   */
+  initDocument = async function (): Promise<void> {
+    this.debug("initDocument()");
+    await this.spoofDocument();
+    this.callInitProcessor(
+      this.config.defaultStyle,
+      this.config.defaultLocale,
+      this.config.citationByIndex,
+      this.config.citationData
+    );
+  };
+
+  /**
+   * Update all citations based on data returned by the processor.
+   * The update has two effects: (1) the id of all in-text citation
+   * nodes is set to the citationByIndex object; and (2)
+   * citation texts are updated.
+   *
+   * data: An array of elements with the form `[citationIndex, citationText, citationID]`
+   */
+  async setCitations(data: object[]): Promise<void> {
+    this.debug("setCitations()");
+
+    for (var i = 0; i < data.length; i++) {
+      const position = data[i][0];
+      const tag = this.config.citationByIndex[position];
+      const encodedTag = btoa(JSON.stringify(tag));
+      const citationTag = await this.getCitationTagByIndex(position);
+      if (citationTag === "NEW" || citationTag != encodedTag) {
+        await this.setCitationTagAtPosition(position, encodedTag);
+      }
+      await this.insertTextInContentControl(citationTag as string, data[i][1]);
+    }
+
+    // Update citationIdToPos for all nodes
+    const getTotalNumberOfCitation = await this.getTotalNumberOfCitation();
+    for (let i = 0; i < getTotalNumberOfCitation; i++) {
+      var citationID = await this.getCitationTagByIndex(i);
+      if (citationID) {
+        this.config.citationIdToPos[citationID] = i;
+      }
+    }
+  }
+
+  // Word API
+  async insertEmptyContentControl() {
```

Comment:
Since the file is already pretty huge, I would propose to split it in different parts. Maybe something like the following:
- that exposes a few methods that are user-facing, i.e. that can be called almost verbatim from the react compotents. For example, something like `insertBibliography` or `insertCitation`.
- that handles the word-specific things, like creating and modifying the content controls etc
- that handles the interaction with the citeproc web worker

What do you think?